src/EditorFeatures/CSharpTest/AddParameter/AddParameterTests.cs

```diff
@@ -2587,5 +2589,57 @@ void TestFunc()
     void MyFunc(BaseClass param1, int newparam) { }
 }");
         }
+
+        [WorkItem(44271, "https://github.com/dotnet/roslyn/issues/44271")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddParameter)]
+        public async Task TopLevelStatement()
+        {
+            await TestMissingAsync(@"
+<Workspace>
+    <Project Language=""C#"" LanguageVersion=""Preview"" AssemblyName=""Assembly1"" CommonReferences=""true"">
+        <Document>
+[|local|](1, 2, 3);
+
+void local(int x, int y)
+{
+}
+        </Document>
+    </Project>
+</Workspace>",
```

Comment:
surprised you used the `<workspace>` form here.  Was that necessary?

Reply:
I didn't know it's possible to using XML approach starting with `<Project>` directly. Will simplify.

Reply:
Hm, on second thought, when I removed `<Workspace>` in test  `TopLevelStatement_Nested` below, it started outputting XML instead of just source. I'm not sure how this works. Is it okay to leave `<Workspace>` here?